apps/address-book/app/app-state-reducer.js

```diff
@@ -1,7 +1,13 @@
-const appStateReducer = state => {
+import { getContentHolder } from '../../../shared/lib/utils'
+
+let prevState = {}
+
+const appStateReducer = currentState => {
+  const state = getContentHolder('entries', currentState, prevState)
+  prevState = { ...state }
```

Comment:
Why not just `prevState = state`? Does it need to be a brand new object?

Reply:
As long as the state doesn't get changed later on, it doesn't need to.

In principle, the practice is to create a `newState` object instead of modifying the state. But just to be safe, I did that. Cause if it gets changed, things can get nasty.

Reply:
I suspect that in general we return new state objects too frequently, and that this is part of what causes unwanted refreshes of our apps.